### DISCUSSION TITLE
OSPOs to improve processes/ tools to ease translation contributions?

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#185] [OSPO Discussions]
Sometimes, language barrier might be critical to prevent non-native English-speaking organizations (e.g APAC or European regions) to use / contribute to open source projects. IMHO, contributions to translations are important in the OSPO ecosystem to:

1) Reach diverse audiences
2) Spread education/resources
3) Avoid duplicating efforts

My question is,  **are there any OSPOs working to improve processes/ tools to ease translation contributions?** If so, could you share any best practices?

**Comment by xmulligan** [comment]
The Kubernetes project has a good framework that we could follow:

See https://kubernetes.io/docs/contribute/localization/. Some key parts would be:
a) If you want the Kubernetes documentation localized into a new language, here's what you need to do. Because contributors can't approve their own pull requests, you need at least two contributors to begin a localization. All localization teams must be self-sustaining. The Kubernetes website is happy to host your work, but it's up to you to translate it and keep existing localized content current.
b) Localizing all of the Kubernetes documentation is an enormous task. It's okay to start small and expand over time. When you start a new localization, you must localize all the [minimum required content](https://kubernetes.io/docs/contribute/localization/#minimum-required-content) before the Kubernetes project can publish your changes to the live website.
c) Localizing the README and CoC

Create a README ([Kubernetes example](https://github.com/kubernetes/website/blob/main/README.md)) that tells people how to build, test, and troubleshoot the documentation.

Decide upon a set of minimum required content for localized documentation to be mereged into master. I would suggest the Cilium Overivew, Cilium Installation, Community, and Development Guide sections.

Make sure the docs fall back to English rather than 404 when a page has not been translated yet.

Reference

These two talks from Zach Corleissen are also helpful to understand how Kubernetes did it:

[Multilingual Kubernetes: The kubernetes.io stack, how we got there, and what it took to get there](https://archive.fosdem.org/2019/schedule/event/multikuber/)

[Localize your open source documentation: a Kubernetes case study](https://www.youtube.com/watch?v=0k9UkMrMBYM&t=1s)

**Comment by caniszczyk** [comment]
A few options used by others out there:

http://zanata.org - Openstack, Red Hat related communities etc (also open source)
https://weblate.org
https://pontoon.mozilla.org/
https://www.transifex.com (more commercial style offering)
